mypy/newsemanal/semanal_namedtuple.py

```diff
@@ -189,15 +189,17 @@ def check_namedtuple(self,
         else:
             default_items = {}
         info = self.build_namedtuple_typeinfo(name, items, types, default_items)
-        # Store it as a global just in case it would remain anonymous.
-        # (Or in the nearest class if there is one.)
-        self.store_namedtuple_info(info, var_name or name, call, is_typed)
+        if var_name:
+            self.store_namedtuple_info(info, var_name, call, is_typed)
+        if name != var_name:
```

Comment:
This check is too subtle. Maybe also explicitly check whether we are in a function scope?

Reply:
OK. (although it will be technically redundant, it is probably more robust)